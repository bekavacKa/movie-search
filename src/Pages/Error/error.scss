@import '../../Assets/scss/mixins.scss';
@import '../../Assets/scss/variables.scss';

.error{
    height: 100vh;
    @include my-flex(center, column, center );
    h2{
        font-size: 4rem;
        margin-bottom: 30px;
    }
    p{
        font-size: 2rem;
        margin-bottom: 30px;
    }
    .error-btn{
        @include my-btn-style;
        background-color: $firstFontColor;
        color: $secondBackColor;
        font-weight: 500;
        &:hover{
            color: $firstFontColor;
        }
    }
}