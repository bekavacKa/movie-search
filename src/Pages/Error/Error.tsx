import React from 'react';
import { Link } from 'react-router-dom';
import routes from '../../Config/routes';
import './error.scss';

function Error() {
  return (
    <div className='error'>
        <h2>
            ERROR 
        </h2>
        <p>
            We can't find the page, or it doesn't exist !
        </p>
        <Link to={routes.HOME.url}>
            <button className='error-btn'>Back to {routes.HOME.name}</button>
        </Link>
    </div>
  )
}

export default Error