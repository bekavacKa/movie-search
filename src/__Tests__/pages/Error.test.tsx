import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Error from '../../Pages/Error/Error';

test('Snapshot test', () => {
  const tree = renderer.create(
    <Router >
            <Error />
        </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
