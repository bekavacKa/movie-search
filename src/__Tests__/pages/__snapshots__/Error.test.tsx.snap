// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="error"
>
  <h2>
    ERROR
  </h2>
  <p>
    We can't find the page, or it doesn't exist !
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="error-btn"
    >
      Back to 
      Home
    </button>
  </a>
</div>
`;
